packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -309,10 +270,18 @@ export async function deployArmTemplates(ctx: SolutionContext): Promise<Result<v
   });
   try {
     result = await doDeployArmTemplates(ctx);
-    ctx.telemetryReporter?.sendTelemetryEvent(SolutionTelemetryEvent.ArmDeployment, {
-      [SolutionTelemetryProperty.Component]: SolutionTelemetryComponentName,
-      [SolutionTelemetryProperty.Success]: SolutionTelemetrySuccess.Yes,
-    });
+    if (result.isOk()) {
+      ctx.telemetryReporter?.sendTelemetryEvent(SolutionTelemetryEvent.ArmDeployment, {
+        [SolutionTelemetryProperty.Component]: SolutionTelemetryComponentName,
+        [SolutionTelemetryProperty.Success]: SolutionTelemetrySuccess.Yes,
+      });
+    } else {
+      sendErrorTelemetryThenReturnError(
+        SolutionTelemetryEvent.ArmDeployment,
+        result.error,
+        ctx.telemetryReporter
+      );
+    }
   } catch (error) {
     result = err(
       returnUserError(
```

Comment:
L287 may need to be changed to SystemError, since user error ( ARM deployment error ) are handled in L279.

Reply:
Sure, it is fixed, thanks.